ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilder.java

```diff
@@ -316,229 +74,166 @@ private static PhysicalSchema getPhysicalSchema(final SourceStep<?> streamSource
+            keyGenerator, streamSource.getPseudoColumnVersion()));
 
-    return new RegisterSchemaCallback(buildContext.getServiceContext().getSchemaRegistryClient());
+    return transformed.mapValues(row -> row, materialized);
   }
 
-  /**
-   * This code mirrors the logic that generates the name for changelog topics
-   * in kafka streams, which follows the pattern:
-   * <pre>
-   *    applicationID + "-" + stateStoreName + "-changelog".
-   * </pre>
-   */
-  private static String changelogTopic(
+  @Override
+  Materialized<GenericKey, GenericRow, KeyValueStore<Bytes, byte[]>> buildTableMaterialized(
+      final SourceStep<KTableHolder<GenericKey>> source,
       final RuntimeBuildContext buildContext,
-      final String stateStoreName
+      final MaterializedFactory materializedFactory,
+      final Serde<GenericKey> sourceKeySerde,
+      final Serde<GenericRow> valueSerde
   ) {
-    return buildContext.getApplicationId()
-        + "-"
-        + stateStoreName
-        + "-changelog";
-  }
 
-  private static TimestampExtractor timestampExtractor(
-      final KsqlConfig ksqlConfig,
-      final LogicalSchema sourceSchema,
-      final Optional<TimestampColumn> timestampColumn,
-      final SourceStep<?> streamSource,
-      final RuntimeBuildContext buildContext
-  ) {
-    final TimestampExtractionPolicy timestampPolicy = TimestampExtractionPolicyFactory.create(
-        ksqlConfig,
-        sourceSchema,
-        timestampColumn
-    );
+    final String stateStoreName = SourceBuilderUtils.tableChangeLogOpName(source.getProperties());
 
-    final Optional<Column> tsColumn = timestampColumn.map(TimestampColumn::getColumn)
-        .map(c -> sourceSchema.findColumn(c).orElseThrow(IllegalStateException::new));
+    final PhysicalSchema physicalSchema = getPhysicalSchemaWithKeyAndPseudoCols(source);
 
-    final QueryContext queryContext = streamSource.getProperties().getQueryContext();
+    final QueryContext queryContext = addMaterializedContext(source);
 
-    return timestampPolicy.create(
-        tsColumn,
-        ksqlConfig.getBoolean(KsqlConfig.KSQL_TIMESTAMP_THROW_ON_INVALID),
-        buildContext.getProcessingLogger(queryContext)
+    final Serde<GenericRow> valueSerdeToMaterialize = getValueSerde(
+        buildContext,
+        source,
+        physicalSchema,
+        queryContext
     );
-  }
 
-  private static <K> Consumed<K, GenericRow> buildSourceConsumed(
-      final SourceStep<?> streamSource,
-      final Serde<K> keySerde,
-      final Serde<GenericRow> valueSerde,
-      final Topology.AutoOffsetReset defaultReset,
-      final RuntimeBuildContext buildContext,
-      final ConsumedFactory consumedFactory) {
-    final TimestampExtractor timestampExtractor = timestampExtractor(
-        buildContext.getKsqlConfig(),
-        streamSource.getSourceSchema(),
-        streamSource.getTimestampColumn(),
-        streamSource,
-        buildContext
+    final Serde<GenericKey> keySerde = buildKeySerde(
```

Comment:
nit: inconsistent naming compared to `valueSerdeToMaterialize` on the line above. (Same for in `buildWindowedTableMaterialized()` below.)